bundle/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/EnhancedSelection.php

```diff
@@ -6,21 +6,35 @@
 
 use Doctrine\DBAL\Connection;
 use Doctrine\DBAL\Query\QueryBuilder;
-use Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion;
+use Ibexa\Contracts\Core\Repository\Values\Content\Query\CriterionInterface;
 use Ibexa\Core\Base\Exceptions\InvalidArgumentException;
 use Ibexa\Core\Search\Legacy\Content\Common\Gateway\CriteriaConverter;
 use Ibexa\Core\Search\Legacy\Content\Common\Gateway\CriterionHandler\FieldBase;
 use Netgen\Bundle\EnhancedSelectionBundle\API\Repository\Values\Content\Query\Criterion\EnhancedSelection as EnhancedSelectionCriterion;
 
 final class EnhancedSelection extends FieldBase
 {
-    public function accept(Criterion $criterion): bool
+    public function accept(CriterionInterface $criterion): bool
     {
         return $criterion instanceof EnhancedSelectionCriterion;
     }
 
-    public function handle(CriteriaConverter $converter, QueryBuilder $queryBuilder, Criterion $criterion, array $languageSettings): string
-    {
+    /**
```

Comment:
superflous PHPDoc. This method does not throw anything and all @param and @return are already declared in method signature itself.

Reply:
That's fine.

As for @throws, we should not specify those if the method does not **directly** throw. Otherwise, we might put every throw on every method ever :D